apps/bfd-pipeline/bfd-pipeline-rif-load/src/main/java/gov/cms/bfd/pipeline/rif/load/RifLoaderIdleTasks.java

```diff
@@ -0,0 +1,355 @@
+      for (Future<Boolean> task : tasks) {
+        isDone = task.get(2 * TASK_TIME_LIMIT_MILLIS, TimeUnit.MILLISECONDS) && isDone;
+      }
+    } catch (TimeoutException ex) {
+    } catch (ExecutionException ex) {
+    } catch (InterruptedException ex) {
+    }
+    return isDone;
+  }
+
+  /**
+   * Fixup a batch of Beneficiaries. Update the beneficary metrics.
+   *
+   * @param em a {@link EntityManager} setup for a transaction
+   * @return true if done with all fixups
+   */
+  public Boolean fixupBeneficiaryBatch(final EntityManager em, final Instant startTime) {
+    LOGGER.debug("Start fixing up a Beneficiary batch");
+    boolean isDone = true;
+    // Use a cursor, measures slightly faster than
+    try (ScrollableResults itemCursor =
+        em.unwrap(Session.class)
+            .createQuery(SELECT_UNHASHED_BENFICIARIES)
+            .setFetchSize(BATCH_COUNT)
+            .scroll(ScrollMode.SCROLL_INSENSITIVE)) {
+      int count = 0;
+      while (inPeriod(startTime, TASK_TIME_LIMIT) && itemCursor.next()) {
+        final Beneficiary beneficiary = (Beneficiary) itemCursor.get(0);
+        beneficiary
+            .getMedicareBeneficiaryId()
+            .ifPresent(
+                mbi -> {
+                  if (mbi.isEmpty()) return;
+                  final String mbiHash =
+                      "xxx"; // RifLoader.computeMbiHash(options, secretKeyFactory, mbi);
+                  beneficiary.setMbiHash(Optional.of(mbiHash));
+                });
+        isDone = itemCursor.isLast();
+        // Write to the DB in batches
+        if (++count % BATCH_COUNT == 0) {
+          em.flush();
+          em.clear();
+        }
+      }
+      beneficaryMeter.mark(count);
+      LOGGER.debug("Finished fixing up a Beneficiary batch: {}", count);
+    }
+    return isDone;
+  }
+
+  /**
+   * Fixup a batch of BeneficiaryHistory. Update the history metrics.
+   *
+   * @param em a {@link EntityManager} setup for a transaction
+   * @return true if done with all fixups
+   */
+  public Boolean fixupHistoryBatch(final EntityManager em, final Instant startTime) {
+    LOGGER.debug("Start fixing up a History batch");
+    boolean isDone = true;
+    try (ScrollableResults itemCursor =
+        em.unwrap(Session.class)
+            .createQuery(SELECT_UNHASHED_HISTORIES)
+            .setFetchSize(BATCH_COUNT)
+            .scroll(ScrollMode.FORWARD_ONLY)) {
+      int count = 0;
+      while (inPeriod(startTime, TASK_TIME_LIMIT) && itemCursor.next()) {
+        final BeneficiaryHistory beneficiary = (BeneficiaryHistory) itemCursor.get(0);
+        beneficiary
+            .getMedicareBeneficiaryId()
+            .ifPresent(
+                mbi -> {
+                  if (mbi.isEmpty()) return;
+                  final String mbiHash =
+                      "xxx"; // RifLoader.computeMbiHash(options, secretKeyFactory, mbi);
+                  beneficiary.setMbiHash(Optional.of(mbiHash));
```

Comment:
There may be some amount of this logic that can be shared between the methods updating the beneficiary and beneficiaryHistory records.